scanner/client/client.go

```diff
@@ -0,0 +1,172 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"github.com/cenkalti/backoff/v3"
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	v4 "github.com/stackrox/rox/generated/internalapi/scanner/v4"
+	"github.com/stackrox/rox/pkg/clientconn"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// ScannerClient provides a client-side interface for Stackrox Scanner.
+type ScannerClient interface {
+	// IndexContainerImage scans and index each layer of the referenced container image.
+	IndexContainerImage(context.Context, name.Digest, authn.Authenticator) (*v4.IndexReport, bool, error)
+	// ContainerImageIndexReport returns an index report of a previously scanned container image.
+	ContainerImageIndexReport(context.Context, name.Digest) (*v4.IndexReport, bool, error)
+	// ScanContainerImage index the referenced container image and creates a
+	// VulnerabilityReport from its IndexReport, which is created if it does not
+	// exist.
+	ScanContainerImage(context.Context, name.Digest, authn.Authenticator) (*v4.VulnerabilityReport, error)
+	// Close closes the scanner client connection and underlying resources.
+	Close() error
+}
```

Comment:
Is another method needed performing matching only? (ie: the local scanning use case when a client already has an IndexReport or `scanner.v4.Contents`)

Reply:
> Is another method needed for performing matching only? 

Seems like you know the answer already. I decided to not add it right now, for simplicity.